src/modules/filesystem.ts

```diff
@@ -0,0 +1,109 @@
+import { dataDirectory, imagesDirectory } from '../config';
+import keyv from 'keyv';
+import { promises as fs, existsSync } from 'fs';
+import mimeTypes from 'mime-types';
+import path from 'path';
+import sharp from 'sharp';
+
+const cache = new keyv();
+
+export async function getTypes(): Promise<string[]> {
+  const found = await cache.get('types');
+  if (found) return found;
+
+  const types = await fs.readdir(dataDirectory(''));
+
+  await cache.set('types', types);
+  console.log('added types to cache');
+
+  return types;
+}
+
+export async function getAvailableEntities(
+  type: string,
+): Promise<string[] | null> {
+  const found = await cache.get(`data-${type}`);
+  if (found) return found;
+
+  const exists = existsSync(dataDirectory(type));
+  if (!exists) throw new Error(`Type ${type} not found`);
+
+  const entities = await fs.readdir(dataDirectory(type));
+  await cache.set(`data-${type}`, entities);
```

Comment:
Is it possible to cache it for contain amount of time? I think its not needed to cache it indefinitely but store only frequently used data to reduce ram usage.